app/main.py

```diff
@@ -2,0 +1,19 @@
+from app.errors import (
+    VaccineError,
+    NotWearingMaskError
+)
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    count_of_mask = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            count_of_mask += 1
+    if count_of_mask:
+        return f"Friends should buy {count_of_mask} masks"
```

Comment:
Code Efficiency: This part of the code is not necessary if you return the message right away when you catch a 'NotWearingMaskError'.